opensrp-chw-anc/src/main/java/org/smartregister/chw/anc/interactor/BaseAncRegisterInteractor.java

```diff
@@ -54,6 +54,51 @@ public BaseAncRegisterInteractor() {
         this(new AppExecutors());
     }
 
+    public static void saveVaccineEvents(JSONArray fields, String baseID, String dob) {
+
+        JSONObject vaccinesAtBirthObject = org.smartregister.util.JsonFormUtils.getFieldJSONObject(fields, DBConstants.KEY.VACCINES_AT_BIRTH);
+        JSONArray vaccinesAtBirthArray = vaccinesAtBirthObject != null ? vaccinesAtBirthObject.optJSONArray(DBConstants.KEY.OPTIONS) : null;
+
+        if (vaccinesAtBirthArray != null) {
+            for (int i = 0; i < vaccinesAtBirthArray.length(); i++) {
+                JSONObject currentVaccine = vaccinesAtBirthArray.optJSONObject(i);
+                if (currentVaccine != null && currentVaccine.optBoolean(JsonFormUtils.VALUE) && !currentVaccine.optString(JsonFormUtils.KEY).equals("chk_none")) {
+                    String VaccineName = currentVaccine.optString(JsonFormUtils.KEY).equals("chk_bcg") ? "bcg" : "opv_0";
+                    VisitUtils.savePncChildVaccines(VaccineName, baseID, vaccinationDate(dob));
+                }
+            }
+
+        } else {
+            saveVaccines(fields, baseID);
+        }
+    }
+
+    public static Date vaccinationDate(String vaccineDate) {
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy", Locale.US);
+        try {
+            return formatter.parse(vaccineDate);
+        } catch (ParseException e) {
+            Timber.e(e);
+        }
+        return null;
+    }
+
+    private static void saveVaccines(JSONArray fields, String baseID) {
+        String[] vaccines = {"bcg_date", "opv0_date"};
```

Comment:
Move string literals to constants class.

Reply:
Done ✅